### DISCUSSION TITLE
how to get the url of image where image is stored?

### DISCUSSION THREAD

**Discussion by muhammadahmad665** [onevcat/Kingfisher#1924] [General]
my image is set into cache by kingfisher I need the URL where this image is stored how to know that?
like FileURL?
thanks

**Comment by onevcat** [comment]
Maybe [this](https://github.com/onevcat/Kingfisher/blob/master/Sources/Cache/DiskStorage.swift#L302-L314) and you can use it like:

```swift
let url = ImageLoader.sampleImageURLs[indexPath.row]
let fileURL = KingfisherManager.shared.cache.diskStorage.cacheFileURL(forKey: url.cacheKey)
```

But be careful, it just gives you a URL where the image "should" be in if it was cached. Remember to check the file existence every time before you actually use that URL to do your things.
